src/picomsg/src/oper_db_writer.cpp

```diff
@@ -114,3 +205,308 @@ bool oper_db_writer_set_terminator_vrf_tunnel_counter(const uint32_t terminator_
 
 	return true;
 }
+
+bool oper_db_writer_del_terminator_vrf_tunnel_counter(const uint32_t terminator_id, const char *tenant_id,
+                                                const char *account_id, const char *vrf_name) {
+	auto oper_db_client = picomsg_get_oper_db_client();
+
+	std::string terminator_id_str = std::to_string(terminator_id);
+
+	const char* keys[] = { terminator_id_str.c_str(), tenant_id, account_id, vrf_name };
+
+	auto db_obj = oper_db_client_create_db_object(*oper_db_client, LIBOPER_DB_CLIENT_CLASS_ID_VRF_TUNNELS_NUMBER_IN_TERMINATOR, keys);
+	if (! db_obj) {
+		PICOMSG_LOG(PICOMSG_LOG_ERR, "failed to create oper DB object for class ID LIBOPER_DB_CLIENT_CLASS_ID_VRF_TUNNELS_NUMBER_IN_TERMINATOR");
+		return false;
+	}
+
+	if (! (*oper_db_client)->del((*oper_db_client).get(), db_obj.release(), 0)) {
+		PICOMSG_LOG(PICOMSG_LOG_ERR, "failed to delete LIBOPER_DB_CLIENT_CLASS_ID_VRF_TUNNELS_NUMBER_IN_TERMINATOR keys {%u, %s, %s, %s} from oper DB",
+		terminator_id, tenant_id, account_id, vrf_name);
+		return false;
+	}
+
+	return true;
+}
+bool oper_db_writer_set_terminator_tunnel_items(const char *vrf_name, const char *device_id) {
+	auto oper_db_client = picomsg_get_oper_db_client();
+	const char* keys[] = {vrf_name, device_id};
+
+	auto db_obj = oper_db_client_create_db_object(*oper_db_client, LIBOPER_DB_CLIENT_CLASS_ID_TUNNEL_ITEM, keys);
+	if (! db_obj) {
+		PICOMSG_LOG(PICOMSG_LOG_ERR, "failed to create oper DB object for class ID LIBOPER_DB_CLIENT_CLASS_ID_TUNNEL_OPER_DATA");
```

Comment:
`LIBOPER_DB_CLIENT_CLASS_ID_TUNNEL_ITEM`, not `LIBOPER_DB_CLIENT_CLASS_ID_TUNNEL_OPER_DATA`